GitManual: Adding support for querying git commands with a single option passed

samskeller opened the pull request:
Fixes #309. Users can now use this Instant Answer with not just the basic git command (`git stash`, `git commit`) but also when they add any single option to the command (`git stash save`, `git commit -m`). When they add a single option, the text displayed should be a description of that command used with that option.

IA Page: https://duck.co/ia/view/git_manual

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

GuiltyDolphin commented:
@samskeller Thanks! Looks good from a quick look through, @moollaza can we get this on Beta?

moollaza commented:
@samskeller looks like `\n` is being treated literally, instead of converted to `\\n` when written to the file.

Take a look at line 2 in ouput.txt -- it's the continuation of the first line -- that should all be on one line 👍

samskeller commented:
@moollaza good catch. I've replaced the `\n` characters with `\\n` and then also replaced `\t` with four spaces `    ` like we do for the usage. Let me know how it looks!

moollaza commented on lib/fathead/git_manual/parse.py:
```diff
@@ -16,24 +16,27 @@ class Command(object):
+        if usage_cleaned:
+            # Make the abstract have the description as well as a code block
+            # for the usage of the command
+            abstract = '{}\n<pre><code>{}</pre></code>'.format(self.description,
```
@samskeller looks like `\n` should be `\\n` here

samskeller replied:
@moollaza ah sorry about this -- I thought you had only wanted to strip newline characters from new code related to this feature -- I didn't realize this was causing problems in production as well for the basic git commands

moollaza commented on lib/fathead/git_manual/parse.py:
```diff
@@ -16,24 +16,27 @@ class Command(object):
-        abstract = '{}\n<pre><code>{}</pre></code>'.format(self.description,
-                                                           usage_cleaned)
+        if usage_cleaned:
+            # Make the abstract have the description as well as a code block
```
@samskeller now that we have Fathead template elements, we should be using them for this IA.

Please make sure the `abstract` is wrapped in a `<section class="prog__container">` tag, and that the `description` text is wrapped in a `<p>`

This will help to space out all the content so it's a bit easier to read 👍 

More info here: https://docs.duckduckhack.com/resources/fathead-overview.html#formatting-the-abstract

Note: It says to use a `div` tag, it should actually be a `section` -- I've pushed a change to the docs and am waiting for them to be redeployed 👍

moollaza commented:
@samskeller thanks! This is looking better 👍 

Left a couple comments about a literal newline before the code blocks that's breaking the output. 

Also, while we're working on this, we should update it to use the new template elements

samskeller commented:
@moollaza in  `7440769` I've fixed the newline character issue and wrapped the abstract and description in `<section>` and `<p>` tags respectively, but I was a bit unsure about what you were saying about template elements. Is there a guide for how these should be integrated into Fathead answers? I was having trouble figuring out what I was supposed to do -- I looked for other Fatheads that are using templates so I could have an example but couldn't find any. Also, does the ["item" template](https://docs.duckduckhack.com/frontend-reference/template-groups.html#my-results-are-mainly-text-with-a-possible-icon-or-logo) make the most sense here?

moollaza commented:
> but I was a bit unsure about what you were saying about template elements. Is there a guide for how these should be integrated into Fathead answers?

@samskeller using the `<section>` and `<p>` tag's is what I meant by the "template elements".

We've added a few CSS classes and additional CSS to better handle the layout of Fatheads when their content is wrapped in a `section` with the `prog__container` class 👍 

You should be good for now, I'll take another look at this soon 😄

samskeller commented:
@moollaza ahh okay cool -- well just let me know if there's anything else I should adjust! Thanks

samskeller commented:
@moollaza friendly bump!

moollaza commented:
@samskeller Thanks! This LGTM :+1: 

This is live on Beta: https://moollaza.duckduckgo.com/?q=git+am+3&t=hq&ia=about

Due to some internal stuff, the triggering is a little funny right now. We replace `-` with a space when it's between words, or remove it when it leads/trails. 

So `git am -u` isn't going to work until we fix that. For now it'll need to be `git am u`
